include/fastdds/rtps/resources/ResourceEvent.h

```diff
@@ -27,109 +28,133 @@
+
+    /*!
+     * @brief This method removes a TimedEventImpl object in case it is waiting to be processed by ResourceEvent's
+     * internal thread.
+     *
+     * This method has to be called before deleting the TimedEventImpl object.
+     * This method cancels any operation of the timer.
+     * Then it avoids the situation of the execution thread calling the event handler when it was removed previously.
+     * @param event TimedEventImpl object that will be deleted and we have to be sure all its operations are cancelled.
+     */
+    void unregister_timer(
+            TimedEventImpl* event);
+
+    /*!
+     * @brief This method notifies to ResourceEvent that the TimedEventImpl object has operations to be scheduled.
+     *
+     * These operations can be the cancellation of the timer or starting another async_wait.
+     * @param event TimedEventImpl object that has operations to be scheduled.
+     */
+    void notify(
+            TimedEventImpl* event);
+
+    /*!
+     * @brief This method notifies to ResourceEvent that the TimedEventImpl object has operations to be scheduled.
+     *
+     * These operations can be the cancellation of the timer or starting another async_wait.
+     * @note Non-blocking call version of the method.
+     * @param event TimedEventImpl object that has operations to be scheduled.
+     * @param timeout Maximum blocking time of the method.
+     */
+    void notify(
+            TimedEventImpl* event,
+            const std::chrono::steady_clock::time_point& timeout);
+
+private:
+
+    //! Warns the internal thread can stop.
+    std::atomic<bool> stop_{ false };
+
+    //! Protects internal data.
+    TimedMutex mutex_;
+
+    //! Used to warn about changes on allow_vector_manipulation_.
+    TimedConditionVariable cv_manipulation_;
+
+    //! Flag used to allow a thread to manipulate the timer collections when the execution thread is not using them.
+    bool allow_vector_manipulation_ = true;
+
+    //! Used to warn there are new TimedEventImpl objects to be processed.
+    TimedConditionVariable cv_;
+
+    //! The total number of created timers.
+    size_t timers_count_ = 0;
+
+    //! Collection of events pending update action.
+    std::vector<TimedEventImpl*> pending_timers_;
+
+    //! Collection of registered events waiting completion.
+    std::vector<TimedEventImpl*> active_timers_;
+
+    //! Current time as seen by the execution thread.
+    std::chrono::steady_clock::time_point current_time_;
+
+    //! Execution thread.
+    std::thread thread_;
+
+    /*!
+     * @brief Registers a new TimedEventImpl object in the internal queue to be processed.
+     * Non thread safe.
+     * @param event Event to be added in the queue.
+     * @return True value if the insertion was successful. In other case, it return False.
+     */
+    bool register_timer_nts(
+            TimedEventImpl* event);
+
+    //! Method called by the internal thread.
+    void run_io_service();
```

Comment:
As we have removed the dependency on asio's io_service, it may be a good idea to change the name of this method.